### PR TITLE
Textsearch api for geocode location details for search query

### PR DESCRIPTION
**Pull request by khusbuchandra**
## Issue Number:
  #186
## Issue Description:
Geocode information for input query
### Summary of solution:
1. /search/textSearch api exposed for input query to fetch result.
2. Response contains the geocode details.
3. Ambiguous address search available now like 'pizza hut' would return nearby store based on user location.

### Can this issue be closed?
No
### Should any new issues be added as a result of this solution?
No
### Have you named your branch in a descriptive way? Remember to name your branch in a unique and descriptive manner in order to properly reflect the issue or feature.
textsearch-api

### CODE REVIEW

**Comment by khusbuchandra** [comment]
Implemented on Search submit functionality:

- Set maps position to the search result 

- Puts a marker on the input query result position 

- Onclick of marker it displays the name & formatted address.

To do: Implement selection of prediction to update map to location of selection and add marker at the location. Currently it logs the selected prediction

File: client/src/components/Search/index.js [hunk 1]
```diff
@@ -34,15 +36,52 @@ export default class Search extends Component {
...
+   *
+   * @param {*} placeDetail 
+   */
+  onPlaceFound(placeDetail) {
+    this.props.map.setZoom(16);
+    const location = placeDetail.geometry.location;
+    const userMarker = L.marker(location).addTo(this.props.map)
+    .bindPopup(`<b>${placeDetail.name} </b>${placeDetail.formatted_address}`);
+    this.props.map.setView(location, 16);
+    //ToDO : add to state , this location should be separate to user location
```
Comment (jkwening) [inline client/src/components/Search/index.js]: additionally, we only want one instance of this marker, if its not null, we should remove the current marker from the map and replace with new one per user selection. Currently, each search submit persists on the map.

File: client/src/components/Search/index.js [hunk 2]
```diff
@@ -34,15 +36,52 @@ export default class Search extends Component {
...
-    alert('A query was submitted: ', this.state.value);
     event.preventDefault();
+    axios.post(`${API_SERVER}/search/textsearch`, {
+      input: this.state.value,
+      lat: this.props.position.lat,
+      lng: this.props.position.lng,
+      }).then((response) => { 
+        if(response.data.status == OK_Status)
+        {
+          let [searchResult] = response.data.results;
```
Comment (jkwening) [inline client/src/components/Search/index.js]: this should be changed to 'const' since `searchResult` is never modified within this score

Comment (jkwening) [reply client/src/components/Search/index.js]: We can create a separate issue for this if it seems tricky to figure out but I ran into an instance were the marker generated for a location in another state. It didn't send me to the location of the first prediction as in other cases. I think we may need to be careful in assuming the first data in the textSearch result array maps is based on proximity to user location.

File: controllers/google-api-controller.js [hunk 3]
```diff
@@ -393,9 +393,63 @@ exports.geocode = (appReq, appRes) => {
...
+*
+* Where output may be either of the following values:
+* - json (recommended) indicates output in JavaScript Object Notation (JSON)
+* - xml indicates output as XML
+*/
+/**
+* @description Handles request for search input query to get back search results and its geocode details
+*
+* @api {POST} /search/textSearch
+* @apiSuccess 200 {JSON} With two root elements:
```
Comment (jkwening) [inline controllers/google-api-controller.js]: The docs needs a little clean up - seems to have information from what you copied from.

File: client/src/components/Search/index.js (hunk 2 above)
Comment (khusbuchandra) [reply client/src/components/Search/index.js]: As per the doc, it states "If you specify a location parameter, you must also specify a radius parameter. radius — Defines the distance (in meters) within which to bias place results. The maximum allowed radius is 50 000 meters. Results inside of this region will be ranked higher than results outside of the search circle; however, prominent results from outside of the search radius may be included." 
We don't have a value specified for radius as part of request currently. Do you think that would help resolve the issue you faced? Shall we default radius to certain value?

Comment (jkwening) [reply client/src/components/Search/index.js]: @khusbuchandra yeah that is probably the best solution. I'll read up more on the specs for text search service. Please create an issue for this for milestone MVP 2.0 as a reminder of something to dig into further.

Depending on how far we go with this app and its feature, we may dictate the radius depending on mode of transportation selected; we probably want different limits for walking vs biking vs driving.

**Pull request merged**